links/gql_http_link/lib/src/link.dart

```diff
@@ -9,6 +9,9 @@ import "package:meta/meta.dart";
 import "./_utils.dart";
 import "./exceptions.dart";
 
+typedef HttpResponseDecoder = Future<Map<String, dynamic>> Function(
```

Comment:
could we change this to `FutureOr<Map<String, dynamic>>` ? then the default encoder could still be sync

Reply:
Return type changed.

Thanks